src/analytics/SegmentAnalyticsService.js

```diff
@@ -158,28 +158,17 @@ class SegmentAnalyticsService {
   identifyAnonymousUser(traits) {
     // if we do not have an authenticated user (indicated by being in this method),
     // but we still have a user id associated in segment, reset the local segment state
-    // This has to be wrapped in the document.readyState logic because the analytics.user()
+    // This has to be wrapped in the analytics.ready() callback because the analytics.user()
     // function isn't available until the analytics.js package has finished initializing.
     return new Promise((resolve, reject) => { // eslint-disable-line no-unused-vars
-      if (document.readyState === 'complete') {
+      global.analytics.ready(() => {
```

Comment:
The code here was written to account for situations where the analytics.ready() event had already been fired by the time identifyAnonymousUser was called.  Do we have a guarantee that global.analytics.ready() is called immediately if analytics is already in a ready state?

Reply:
(Otherwise, this will effectively do nothing if analytics is already "ready", right?)

Reply:
@davidjoy Good question. It looks like `ready` is implemented such that if it is already in a ready state, the callback function is immediately called: https://github.com/segmentio/analytics.js/blob/master/analytics.js#L678-L687

To further confirm this, I did something like the following:

```jsx
global.analytics.ready(() => {
  console.log('analytics.js is ready');
});

setTimeout(() => {
  global.analytics.ready(() => {
    console.log('ready callback still gets called');

    if (global.analytics.user().id()) {
      global.analytics.reset();
    }
    global.analytics.identify(traits);
    this.hasIdentifyBeenCalled = true;
    resolve();
  });
}, 5000);
```

where both `console.log` lines are called.

Reply:
Perfect!